### PR TITLE
Validation tests for early eval errors in subgroup builtins

### PR DESCRIPTION
**Pull request by alan-baker**
* subgroupBroadcast and quadBroadcast id must be in range [0, 128)
* subgroupShuffleUp, subgroupShuffleDown, subgroupShuffleXor mask/delta must be less than 128 for const/override

Note: the shuffle parameters are required to be u32, so negative values are impossible.

Issue: #

<hr>

**Requirements for PR author:**

- [ ] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [ ] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [ ] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [ ] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/shader/validation/expression/call/builtin/subgroupShuffle.spec.ts [hunk 1]
```diff
@@ -94,6 +94,57 @@ g.test('early_eval')
...
   });
 
+g.test('param2_early_eval')
+  .desc('Ensures delta/mask parameters must be in the range [0, 128) for const and override')
+  .params(u =>
+    u
+      .combine('op', kOps)
+      .filter(t => {
+        return t.op !== 'subgroupShuffle';
+      })
```
Comment (alan-baker) [inline src/webgpu/shader/validation/expression/call/builtin/subgroupShuffle.spec.ts:104]: oops this shouldn't be here.
```suggestion
```

File: src/webgpu/shader/validation/expression/call/builtin/subgroupShuffle.spec.ts [hunk 2]
```diff
@@ -94,6 +94,54 @@ g.test('early_eval')
     t.expectCompileResult(t.params.stage === 'runtime', code);
   });
 
+g.test('param2_early_eval')
+  .desc('Ensures delta/mask parameters must be in the range [0, 128) for const and override')
+  .params(u =>
+    u
+      .combine('op', kOps)
+      .combine('value', [0, 127, 128] as const)
```
Comment (petermcneeleychromium) [inline src/webgpu/shader/validation/expression/call/builtin/subgroupShuffle.spec.ts:102]: I think subgroupShuffle (not up/down/xor variants) can have a signed value.

Comment (alan-baker) [reply src/webgpu/shader/validation/expression/call/builtin/subgroupShuffle.spec.ts:102]: Good catch, I'll update.

**Review by dneto0 — APPROVED** [review]
LGTM. 
Thanks

**Review by petermcneeleychromium — APPROVED** [review]
lets get this in. I can resolve potential issues on the dawn side in combination with my recent changes.
My concern here is if the Overrides are going to work correctly.

**Review by jrprice — APPROVED** [review]

**Pull request merged**